storage/remote/client.go

```diff
@@ -149,6 +151,17 @@ func (c *Client) Read(ctx context.Context, query *prompb.Query) (*prompb.QueryRe
 	httpReq.Header.Set("User-Agent", userAgent)
 	httpReq.Header.Set("X-Prometheus-Remote-Read-Version", "0.1.0")
 
+	if span := opentracing.SpanFromContext(ctx); span != nil {
```

Comment:
nit: the recommended way to trace remote calls is to create a new child span here, and capture in its tag various useful info like the query.